### DISCUSSION TITLE
[FAIL. Reason: EvmError: Revert]

### DISCUSSION THREAD

**Discussion by hoBabu1** [Cyfrin/foundry-full-course-cu#1700] [Q&A]
I got this error when i am running test for my DSCEngine. Problem is not in testing script, I just called  a simple function and did no checking, it failed . I have attached a screenShot, please help me out .
![Screenshot 2024-05-11 130442](https://github.com/Cyfrin/foundry-full-course-f23/assets/128292819/06349d65-cf27-48ac-ae79-5feab6c73edb)

**Comment by EngrPips** [comment]
Hello @hoBabu1, Please show your entire test file and the full output you get when you run your test.

**Reply by hoBabu1** [reply]
Sure, 
I have changed something in the code that will make more clear whats the issue 

``` //SPDX-License-Identifier:MIT
pragma solidity ^0.8.0;

import {Test,console} from "forge-std/Test.sol";
import {DeployDSC} from "script/DeployDsc.s.sol";
import {DecentralizedStableCoin} from "src/DecentralizedStableCoin.sol";
import {HelperConfig} from "script/HelperConfig.s.sol";

import {DSCBrain} from "src/DSCBrain.sol";

contract DSCBrainTest is Test {
    DeployDSC dscDeployer;
    DSCBrain dscBrain;
    HelperConfig helperConfig;
    DecentralizedStableCoin dscCoin;

    address wethUsdPriceFeed;
    address wbtcUsdPriceFeed;
    address weth;
    address wbtc;
    uint256 deployerKey;

    function setUp() external {
        dscDeployer = new DeployDSC();
        (dscCoin, dscBrain, helperConfig) = dscDeployer.run();
        (wethUsdPriceFeed, wbtcUsdPriceFeed, weth, wbtc, deployerKey) = helperConfig.activeNetworkConfig();
    }
    function testgetValueinUsd() public  {
        uint256 a = 6;
        uint256 b = 6;
        dscBrain.getValueinUsd(weth,5e18);
        assert(a==b);
    }
}
``` 

This is the code of TestScript

here is the error 
``` Ran 1 test for test/mocks/unit/DSCBrainTest.t.sol:DSCBrainTest
[FAIL. Reason: EvmError: Revert] testgetValueinUsd() (gas: 12615)
Traces:
  [12615] DSCBrainTest::testgetValueinUsd()
    ├─ [5458] DSCBrain::getValueinUsd(0xdd13E55209Fd76AfE204dBda4007C227904f0a81, 5000000000000000000 [5e18]) [staticcall]
    │   ├─ [0] 0x694AA1769357215DE4FAC081bf1f309aDC325306::latestRoundData() [staticcall]
    │   │   └─ ← [Stop] 
    │   └─ ← [Revert] EvmError: Revert
    └─ ← [Revert] EvmError: Revert

Suite result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 1.26ms (118.74µs CPU time)

Ran 1 test suite in 34.20s (1.26ms CPU time): 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/mocks/unit/DSCBrainTest.t.sol:DSCBrainTest
[FAIL. Reason: EvmError: Revert] testgetValueinUsd() (gas: 12615) ```


```
I am attaching repo link 
https://github.com/hoBabu1/decentralizedStableCoin ```

waiting for your response

**Reply by hoBabu1** [reply]
![error](https://github.com/Cyfrin/foundry-full-course-f23/assets/128292819/535cb370-b7dd-4fc6-b391-6e321c82c026)

**Reply by EngrPips** [reply]
Thanks for sharing the repo, I will look to see what is wrong but it seem like the process got stopped when you called `latestRoundData()` on `0x694AA1769357215DE4FAC081bf1f309aDC325306`, can you please confirm if `0x694AA1769357215DE4FAC081bf1f309aDC325306` is a correct priceFeed?

**Reply by hoBabu1** [reply]
When i deployed on sepolia testnet its working fine . 
I am not able to deploy ERC mock with fixed price on anvil chain . 
In tutorial Patrick sir was able to send 4 arguments to ERC20Mock
```  ERC20Mock wbtcMock = new ERC20Mock("WBTC", "WBTC", msg.sender, 1000e8);```
but i am not able to fix its decimal place and its price 
``` ERC20Mock wethMock = new ERC20Mock(); ```
because in constructor of ERC20 mock its not taking any argument .

**Reply by EngrPips** [reply]
Yeah, the new version of `ERCMock` does not seem to take any argument. You should still be able to fix the price and decimal as that is done on the `priceFeed` of the token and not the token itself.

**Comment by hoBabu1** [comment]
There is two photo attached to it . I simply modified ERC20 Mock s u can see in the image attached to it and now its working . 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/128292819/c2150d98-2317-4a3b-bc35-f961934c5369)
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/128292819/b9cedad0-fbcc-456a-a58d-6bea7fd0b939)
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/128292819/9afc9943-5efd-4173-b1d4-9d9739091a7c)

**Reply by EngrPips** [reply]
Awesome, I am glad you unblocked yourself.
